packages/react-data-grid/src/ReactDataGrid.js

```diff
@@ -331,26 +329,14 @@ class ReactDataGrid extends React.Component {
     }
   };
 
-  onSelect = (selected: SelectedType) => {
-    if (this.state.selected.rowIdx !== selected.rowIdx
-      || this.state.selected.idx !== selected.idx
-      || this.state.selected.active === false) {
-      let idx = selected.idx;
-      let rowIdx = selected.rowIdx;
-      if (this.isCellWithinBounds(selected)) {
-        const oldSelection = this.state.selected;
-        this.setState({selected: selected}, () => {
-          if (typeof this.props.onCellDeSelected === 'function') {
-            this.props.onCellDeSelected(oldSelection);
-          }
-          if (typeof this.props.onCellSelected === 'function') {
-            this.props.onCellSelected(selected);
-          }
-        });
-      } else if (rowIdx === -1 && idx === -1) {
-        // When it's outside of the grid, set rowIdx anyway
-        this.setState({selected: { idx, rowIdx }});
-      }
+  onSelect = (selected) => {
+    const {selectCell, onCellDeSelected, onCellSelected} = this.props;
+    selectCell(selected);
```

Comment:
no more setting state on each cell selection, this now triggers an action, and any component can subscribe to state change if necessary